src/main/java/net/imglib2/img/cell/CellImgFactory.java

```diff
@@ -164,6 +164,7 @@ public static void verifyDimensions( final long dimensions[] ) throws IllegalArg
 			final T type,
 			final NativeTypeFactory< T, A > typeFactory )
 	{
+		Dimensions.InvalidDimensions.checkAllPositiveOrElseThrow( dimensions );
```

Comment:
The subsequent `verifyDimensions` check is already checking for positive dimensions, and throwing an `IllegalArgumentException` if not.

Reply:
I will look into it.

Reply:
I moved the call to `Dimensions.verifyAllPositive` into `CellImgFactory.verifyDimensions`. We could even extract `verifyDimensions` into the `Dimensions` class.